### PR TITLE
Add interfaces

### PR DESCRIPTION
**Pull request by NataliaPanchuk**

### CODE REVIEW

File: src/Animals/Animals.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import Animals.factory.AnimalFactory;
+import Animals.factory.Generation;
+import interfaces.ToDead;
+import interfaces.ToEat;
+import interfaces.ToMove;
+import interfaces.ToReproduction;
+
+import java.util.List;
+
+public abstract class Animals implements ToDead, ToEat, ToMove, ToReproduction, Generation {
```
Comment (Volodymyr95) [inline src/Animals/Animals.java]: no needed these interfaces: ToDead, ToEat, ToMove, ToReproduction
u can add it to core(parent) class

File: src/Animals/logics/Basic.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    private IslandCell cell;
+
+    public Basic(Island island) {
+        this.island = island;
+    }
+
+    private void launchBasicLogic() {
+        if (object.isDead() && object.getMove() > 0) {
+            object.setMove(0);
+            if (object.getAnimalSpecies() != AnimalSpecies.PLANTS) {
```
Comment (Volodymyr95) [inline src/Animals/logics/Basic.java:28]: magic number

File: src/Animals/logics/Basic.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                                movementLogic.startMove(1, coordinateX, coordinateY, object, earthCell);
+                            }
+                    }
+
+                }
+            }
+        }
+    }
+
+    //}
```
Comment (Volodymyr95) [inline src/Animals/logics/Basic.java]: formatting

File: src/Animals/logics/BreedingLogic.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import Animals.AnimalSpecies;
+
+import java.util.List;
+
+public class BreedingLogic {
+    public int startReproduction(Animals object, IslandCell cell) {
+        if (cell.getMapAnimal().get(object.getAnimalSpecies()).size() < object.getMaxAnimal()) {
+            List<Animals> animals = cell.getMapAnimal().get(object.getAnimalSpecies());
+            for (Animals animal : animals) {
+                if (animal.isDead() && animal.getFullSatiety() > (animal.getSatiety() * 0.9)) {
```
Comment (Volodymyr95) [inline src/Animals/logics/BreedingLogic.java]: magic numbers

File: src/Animals/logics/MovementLogic.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+public class MovementLogic {
+    private int x, y;
+
+    public void startMove(int direction, int corX, int corY, Animals object, IslandCell[][] earthCell) {
+
+        x = corX;
+        y = corY;
+        int steep = object.move();
+
+        if (steep == 0) {
```
Comment (Volodymyr95) [inline src/Animals/logics/MovementLogic.java]: what this logic mean?

File: src/Animals/logics/MovementLogic.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            steep = 1;
+        }
+
+        if (direction == 0) {
+
```
Comment (Volodymyr95) [inline src/Animals/logics/MovementLogic.java:18]: formatting

File: src/Animals/Animals.java (hunk 1 above)
Comment (NataliaPanchuk) [reply src/Animals/Animals.java]: Done

File: src/Animals/logics/Basic.java (hunk 3 above)
Comment (NataliaPanchuk) [reply src/Animals/logics/Basic.java]: Done

File: src/Animals/logics/MovementLogic.java (hunk 6 above)
Comment (NataliaPanchuk) [reply src/Animals/logics/MovementLogic.java:18]: Done

File: src/Animals/logics/Basic.java (hunk 2 above)
Comment (NataliaPanchuk) [reply src/Animals/logics/Basic.java:28]: Done

File: src/Animals/logics/BreedingLogic.java (hunk 4 above)
Comment (NataliaPanchuk) [reply src/Animals/logics/BreedingLogic.java]: Done

File: src/Animals/logics/MovementLogic.java (hunk 5 above)
Comment (NataliaPanchuk) [reply src/Animals/logics/MovementLogic.java]: unnecessary conditions have been removed

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**